client/hotrod-client/src/test/java/org/infinispan/client/hotrod/iteration/AbstractRemoteIteratorTest.java

```diff
@@ -0,0 +1,88 @@
+package org.infinispan.client.hotrod.iteration;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.query.testdomain.protobuf.AccountPB;
+import org.infinispan.commons.marshall.Marshaller;
+import org.infinispan.query.dsl.embedded.testdomain.hsearch.AccountHS;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * @author gfernandes
+ * @since 8.0
+ */
+@SuppressWarnings("unchecked")
+public interface AbstractRemoteIteratorTest {
+
+   default <T> Set<T> setOf(T... values) {
+      return Stream.of(values).collect(Collectors.toSet());
+   }
+
+   default <T> void populateCache(int numElements, Function<Integer, T> supplier, RemoteCache<Integer, T> remoteCache) {
+      IntStream.range(0, numElements).forEach(i -> remoteCache.put(i, supplier.apply(i)));
+   }
+
+   default <T> void assertForAll(Set<T> elements, Predicate<? super T> condition) {
+      assertTrue(elements.stream().allMatch(condition));
+   }
+
+   default AccountHS newAccount(int id) {
+      AccountHS account = new AccountHS();
+      account.setId(id);
+      account.setDescription("description for " + id);
+      account.setCreationDate(new Date());
+      return account;
+   }
+
+   default AccountPB newAccountPB(int id) {
+      AccountPB account = new AccountPB();
+      account.setId(id);
+      account.setDescription("description for " + id);
+      account.setCreationDate(new Date());
+      return account;
+   }
+
+   default Set<Integer> rangeAsSet(int minimum, int maximum) {
+      return IntStream.range(minimum, maximum).boxed().collect(Collectors.toSet());
+   }
+
+   default <K> Set<K> extractKeys(Set<Map.Entry<Object, Object>> entries) {
```

Comment:
Seems this and the next method could be merged into 1 where instead of Set or List it is defined as Collection.